src/shogun/kernel/MaternKernel.cpp

```diff
@@ -0,0 +1,85 @@
+/*
+ * This software is distributed under BSD 3-clause license (see LICENSE file).
+ *
+ * Authors: Gil Hoben
+ */
+
+#include <shogun/kernel/MaternKernel.h>
+
+#include <cmath>
+
+CMaternKernel::CMaternKernel(): CDistanceKernel(), nu(0.0)
+{
+	init();
+}
+
+CMaternKernel::CMaternKernel(int32_t size, float64_t w, float64_t order, CDistance* dist): 
+	CDistanceKernel(size, w, dist), nu(order)
+{
+	ASSERT(distance)
+	SG_REF(distance);
+	init();
+}
+
+CMaternKernel::~CMaternKernel()
+{
+	cleanup();
+	SG_UNREF(distance);
+}
+
+void CMaternKernel::cleanup()
+{
+}
+
+bool CMaternKernel::init(CFeatures* l, CFeatures* r)
+{
+	ASSERT(distance)
+	CDistanceKernel::init(l,r);
+	distance->init(l,r);
+	return init_normalizer();
+}
+
+void CMaternKernel::init()
+{
+	nu = 1.5;
+
+	SG_ADD(&nu, "nu", "order of the Bessel function of the second kind", ParameterProperties::HYPER)
+}
+
+float64_t CMaternKernel::compute(int32_t idx_a, int32_t idx_b)
+{
+	float64_t result;
+
+	float64_t dist = distance->distance(idx_a, idx_b);
+
+	// first we can check if we should use one of the approximations which are cheaper to calculate
+	if (CMath::fequals(nu, 0.5, std::numeric_limits<float64_t>::epsilon()))
```

Comment:
@vigsterkr do we have this somewhere else in the library now?

Reply:
i dont think atm